packages/es-components/src/components/controls/range-selector/RangeSelector.js

```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { noop } from 'lodash';
+
+const RangeSelectorBase = styled.div`
+  width: 100%;
+`;
+const RangeSelectorSlider = styled.div`
+  height: 5px;
+  border-radius: 5px;
+  background: ${props => props.sliderColor};
+  position: relative;
+`;
+const RangeSelectorProgress = styled.div`
+  height: 5px;
+  left: ${props => `${props.minPercentage}%`};
+  right: ${props => `${100 - props.maxPercentage}%`};
+  border-radius: 5px;
+  background: ${props => props.progressColor};
+  position: absolute;
+`;
+const RangeSelectorInputContainer = styled.div`
+  position: relative;
+`;
+
+const RangeSelectorInput = styled.input`
+  position: absolute;
+  top: -5px;
+  height: 5px;
+  width: 100%;
+  background: none;
+  pointer-events: none;
+  -webkit-appearance: none;
+  &:focus {
+    outline: none;
+  }
+  &::-webkit-slider-thumb {
+    height: 17px;
+    width: 17px;
+    border-radius: 50%;
+    pointer-events: auto;
+    -webkit-appearance: none;
+    background: ${props => props.thumbColor};
+  }
+  &::-moz-range-thumb {
+    height: 17px;
+    width: 17px;
+    border: none;
+    pointer-events: auto;
+    -moz-appearance: none;
+    background: ${props => props.thumbColor};
+  }
+`;
+
+function RangeSelector({
+  currentMinValue,
+  currentMaxValue,
+  minValue,
+  maxValue,
+  progressColor,
+  sliderColor,
+  thumbColor,
+  onChange,
+  ...RangeSelectorProps
+}) {
+  const initStateMin = () => {
+    if (currentMinValue < minValue) return 0;
+    if (currentMinValue > maxValue) return 100;
+    return ((currentMinValue - minValue) / (maxValue - minValue)) * 100;
+  };
+  const initStateMax = () => {
+    if (currentMaxValue > maxValue) return 100;
+    if (currentMaxValue < minValue) return 0;
+    return ((currentMaxValue - minValue) / (maxValue - minValue)) * 100;
+  };
+  const [min, setMin] = useState(currentMinValue);
+  const [max, setMax] = useState(currentMaxValue);
```

Comment:
This kind of component is going to be so specific to its implementation, that I think it should be a [Controlled Component](https://legacy.reactjs.org/docs/forms.html#controlled-components). This would allow us to eliminate any internal state, and simply use the `onChange` event to tell our parent what the current min and max are.

An implementation of something like that might be:
```jsx
const [myRange, setMyRange] = useState([1, 10]);
<RangeSelector
  min={1}
  max={10}
  currentMin={myRange[0]}
  currentMax={myRange[1]}
  onChange={setMyRange}
/>
```

Reply:
the  parent will give min, max and set current min and current max.   so it is not internal controlled, 
![image](https://user-images.githubusercontent.com/104922269/225984085-7e66d4f0-5b3d-446b-b059-b3a80658da8f.png)

Reply:
changed from duo range selector to single range selector, removed state in the component.